graph/GuidelinesGraph.md

```diff
@@ -277,6 +278,37 @@ Operation resources MUST have a binding parameter that matches the type of the b
+
+| HTTP Code                 |  Description | Handled At         |
+|---------------------------|--------------|--------------------|
+| 200 OK                    | Indicates that the request was successful, and the response contains the requested data or confirmation.                                 | Service            |
+| 201 Created               | Indicates that a new resource has been successfully created. The `Location` header SHOULD contain the URI of the newly created resource. | Service            |
+| 202 Accepted              | Indicates that the request has been accepted for processing, but the processing is not yet complete. See [Long-Running operations](/graph/patterns/long-running-operations.md). | Service            |
+| 204 No Content            | Indicates that the request was successful, but there is no content to return in the response body.                                       | Service            |
+| 400 Bad Request           | Indicates that the request is malformed or contains invalid parameters. The response SHOULD include details about the error.             | Service            |
+| 401 Unauthorized          | Indicates that the request is unauthenticated. The client MUST authenticate before making the request.                                      | Gateway            |
+| 403 Forbidden             | Indicates that the client is authenticated but does not have sufficient permissions to perform the requested operation.    | Gateway or Service |
+| 404 Not Found             | Indicates that the requested resource could not be found.                                                                                | Gateway or Service |
+| 422 Unprocessable Content | Indicates that a request is syntactically correct but semantically wrong per the business policy, server state or domain rules.          | Service            |
+| 429 Too Many Requests     | Indicates that the client has exceeded the rate limits or throttling thresholds.                                                         | Gateway or Service     |
+| 500 Internal Server Error | Indicates an internal server error. This is a generic error message for unexpected failures.                                             | Service            |
+| 503 Service Unavailable   | Indicates that the service is temporarily unavailable, often due to maintenance or overload.                                             | Gateway or Service |
+
+#### Choosing Between 400 and 422 Response Codes
+
+When designing APIs, it's important to differentiate between client errors that are syntactically incorrect (400 Bad Request) and those that are semantically incorrect (422 Unprocessable Content). The following table provides guidance on when to use each response code, along with examples.
+
+|   | 400 Bad Request      | 422 Unprocessable Content     |
```

Comment:
What about:
/foo/$filter= 'name' eq 'x'
when the service doesn't support name filtering (or insert any other non-preference based request that is valid in the protocol)
Is that also 422, or something else?